### PR TITLE
Message cell adjustments

### PR DESCRIPTION
**Pull request by evsaev**

### CODE REVIEW

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptions.swift [hunk 1]
```diff
@@ -15,6 +15,7 @@ public struct ChatMessageLayoutOptions: OptionSet, Hashable {
 public extension ChatMessageLayoutOptions {
     /// Typically the messages sent by the current user have flipped content
     static let flipped = Self(rawValue: 1 << 0)
+    static let maxWidth = Self(rawValue: 1 << 15)
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptions.swift:18]: Very good idea 👍

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift [hunk 2]
```diff
@@ -293,6 +292,14 @@ class MessageContentView<ExtraData: ExtraDataTypes>: _View, UIConfigProvider {
...
             ]
         }
+
+        constraintsToActivate += fullWidthViews.map { view in
+            view.widthAnchor
+                .pin(equalTo: mainContainer.widthAnchor)
+                // should work with just .streamLow but it doesn't
+                // work for stream preview
+                .with(priority: .streamAlmostRequire)
+        }
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift:302]: That's probably the most interesting part of the PR but not sure `maxWidth` option is valuable for the new attachments approach

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageMetadataView.swift [hunk 3]
```diff
@@ -58,6 +67,9 @@ open class _ChatMessageMetadataView<ExtraData: ExtraDataTypes>: _View, UIConfigP
             timestampLabel.text = nil
         }
         currentUserVisabilityIndicator.isVisible = content?.onlyVisibleForCurrentUser ?? false
+
+        authorLabel.isHidden = content?.isSentByCurrentUser ?? true
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageMetadataView.swift:71]: Nope, this should be decided on the `ChatMessageLayoutOptions` level. For example, in 1:1 channel, the author name is never visible, only when there are more than the current user + 1 person there, we show the names for everyone but the current user.

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift [hunk 4]
```diff
@@ -98,17 +100,26 @@ class MessageContentView<ExtraData: ExtraDataTypes>: _View, UIConfigProvider {
         layoutOptions = options
         
         var constraintsToActivate: [NSLayoutConstraint] = []
+        var fullWidthViews: [UIView] = []
         
         // Main container
         mainContainer.alignment = .axisTrailing
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift:106]: I actually think we can achieve the full-width functionality just by changing this to `.fill`. It probably won't work yet but it should eventually :)

**Review by VojtaStavik — APPROVED** [review]

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift [hunk 5]
```diff
@@ -412,7 +419,7 @@ private extension MessageContentView {
             textView?.isScrollEnabled = false
             textView?.backgroundColor = .clear
             textView?.adjustsFontForContentSizeCategory = true
-            textView?.textContainerInset = .zero
+            textView?.textContainerInset = .init(top: 0, left: 8, bottom: 0, right: 8)
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift:422]: @VojtaStavik  I was thinking about extending `ContainerView` so it's possible to provide custom layout margins for each arranged subview but decided to discuss `ContainerView` with you first and keep it simple in the PR

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift (hunk 4 above)
Comment (evsaev) [reply Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift:106]: The `fill` requires us to rework existing components so they can stretch but still look right

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift (hunk 5 above)
Comment (VojtaStavik) [reply Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift:422]: @evsaev sure, let's talk about the details during auditing 👍

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageMetadataView.swift (hunk 3 above)
Comment (evsaev) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageMetadataView.swift]: Fixed, please have a look.

The fact we don't have access to `cid` from the message brings the complexity 🤔 Why not to change this:
```swift
extension _ChatChannel {
    var isDirectMessageChannel: Bool { cid.id.hasPrefix("!members") }
}
```
with:
```swift
struct _ChatMessage {
    let cid: ChannelId
}
extension ChannelId {
    var isDirectMessageChannel: Bool { id.hasPrefix("!members") }
}
```

Comment (VojtaStavik) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageMetadataView.swift]: @evsaev gotcha. Let's leave it as it, it's not _that_ bad :) Another thing to think about properly during the audit.

**Pull request merged**